.github/workflows/ci-test.yml

```diff
@@ -125,6 +144,314 @@ jobs:
+            echo "::set-output name=should_run::false";
+          else
+            echo "::set-output name=should_run::true";
+          fi;
+      - name: Determine which cluster-test suite to run
+        if: steps.check_ks.outputs.should_run == 'true'
+        # Runs land_blocking if KILL_SWITCH_LAND_BLOCKING_COMPAT activated.
+        # Runs land_blocking_compat otherwise.
+        # Compat needs to find an image with PREV_TAG to test against, otherwise will
+        # create one from BASE_GIT_REV in the next step, flagged by setting BUILD_PREV
+        run: |
+          set +e
+          if ${{ secrets.KILL_SWITCH_LAND_BLOCKING_COMPAT }}; then
+            echo "Compat killswitch activated! Will run land_blocking suite"
+            echo "TEST_COMPAT=0" >> $GITHUB_ENV
+          else
+            echo "Will run land_blocking_compat suite"
+            echo "TEST_COMPAT=1" >> $GITHUB_ENV
+            echo "Finding a previous image tag to test against"
+            .github/actions/land-blocking/find-lbt-images.sh > lbt_images_output.txt
+            if [ $? -ne 0 ]; then
+              echo "BUILD_PREV=1" >> $GITHUB_ENV
+              cat lbt_images_output.txt
+              jq -n \
+                --arg msg "*${{ github.job }}* job in ${{ github.workflow }} workflow failed. Could not find a recent image tag for Compat Test" \
+                --arg url "https://github.com/${{ github.repository }}/actions/runs/${{github.run_id}}" \
+              '{
+                "attachments": [
+                  {
+                    "text": $msg,
+                    "actions": [
+                      {
+                        "type": "button",
+                        "text": "Visit Job",
+                        "url": $url
+                      }
+                    ]
+                  }
+                ]
+              }' > /tmp/payload
+              curl -X POST -H 'Content-type: application/json' -d @/tmp/payload ${{ secrets.WEBHOOK_FLAKY_LAND_BLOCKING_CT }}
+              exit 0
+            else
+              compat_prev_tag=$(tail -1 lbt_images_output.txt)
+              echo "Using previous image tag $compat_prev_tag"
+              echo "PREV_TAG=$compat_prev_tag" >> $GITHUB_ENV
+            fi
+          fi
+          echo "BUILD_PREV=0" >> $GITHUB_ENV
+      - name: Build, tag and push an extra image if it is needed by compat
+        if: steps.check_ks.outputs.should_run == 'true'
+        # If compat needs an extra image
+        # built (e.g. BUILD_PREV -eq 1) then the builds occur in parallel.
+        run: |
+          if [ $BUILD_PREV -eq 1 ]; then
+            compat_prev_tag=land_$BASE_GIT_REV
+            echo "Starting codebuild for $compat_prev_tag"
+            echo "PREV_TAG=$compat_prev_tag" >> $GITHUB_ENV
+            VERSION=$BASE_GIT_REV ADDL_TAG=$compat_prev_tag .github/actions/land-blocking/cti-codebuild.sh &> codebuild-prev.log &
```

Comment:
ah, we are calling codebuild again here.  Building the missing images here without codebuild would be tricky. 

Got an idea to think about - do this check before the `build-images` step, and use it to orchestrate two `build-images` jobs (one for the PR, one for the parent commit), both of which will kick off the LBT job here.

Reply:
May I ask some details here for clear. I understand if we call codebuild here will use another ~20mins to build the code, I agree we can move this step to run as a parallel job with image build which will save this ~20mins. 
But do we wanna deprecate aws codebuild here to use same way as build-images to use build-push instead? I think for this one time image build, aws-codebuild will be much easier